### PR TITLE
Adds open? to workflow state service.

### PR DESCRIPTION
**Pull request by justinlittman**
## Why was this change made? 🤔
The definition of open in the Version Service isn't complete.

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, including data writes to shared file systems, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

Unit

### CODE REVIEW

File: app/controllers/objects_controller.rb [hunk 1]
```diff
@@ -95,7 +95,7 @@ def accession
       updated_cocina_object = VersionService.open(cocina_object: @cocina_object, **version_open_params)
       VersionService.close(druid: updated_cocina_object.externalIdentifier, version: updated_cocina_object.version, **version_close_params.merge(start_accession: false))
     # if this is an existing accessioned object that is currently open, just close it without starting accessionWF
-    elsif VersionService.open?(druid: @cocina_object.externalIdentifier, version: @cocina_object.version)
+    elsif WorkflowStateService.active_version_wf?(druid: @cocina_object.externalIdentifier, version: @cocina_object.version)
```
Comment (justinlittman) [inline app/controllers/objects_controller.rb:98]: `VersionService.open?` was just calling `WorkflowStateService.active_version_wf?`

File: app/controllers/versions_controller.rb [hunk 2]
```diff
@@ -55,7 +55,7 @@ def status
 
     render json: {
       versionId: @version,
-      open: version_service.open_for_versioning?,
+      open: workflow_state_service.open?,
```
Comment (justinlittman) [inline app/controllers/versions_controller.rb:58]: Switching to use WorkflowStateServices definition of `open?` which handles version 1 objects.

File: app/services/version_service.rb [hunk 3]
```diff
@@ -24,12 +24,6 @@ def self.can_open?(druid:, version:, assume_accessioned: false)
     new(druid:, version:).can_open?(assume_accessioned:)
   end
 
-  # @param [String] druid of the item
```
Comment (justinlittman) [inline app/services/version_service.rb:27]: Getting rid of `open?` / `open_for_versioning?` on VersionService. These are handled by WorkflowStateService.

File: app/services/version_service.rb [hunk 4]
```diff
@@ -156,17 +150,11 @@ def retrieve_version_from_preservation
...
-  def open_for_versioning?
-    workflow_state_service.active_version_wf?
-  end
-
   attr_reader :druid, :version
 
   private
 
-  delegate :active_assembly_wf?, :accessioning?, to: :workflow_state_service
+  delegate :active_assembly_wf?, :accessioning?, :open?, to: :workflow_state_service
```
Comment (justinlittman) [inline app/services/version_service.rb:157]: This is the big switch: Changing to a definition of `open?` that accounts for version 1 objects.

File: bin/migrate-cocina [hunk 5]
```diff
@@ -80,7 +80,9 @@ def perform_migrate(migrator_class:, obj:, mode:)
   migrator.migrate
   updated_cocina_object = obj.to_cocina_with_metadata # This validates the cocina object
   Cocina::ObjectValidator.validate(updated_cocina_object) # This validation is performed by UpdateObjectService.
-  raise 'Cannot version' if migrator.version? && !(VersionService.can_open?(druid: updated_cocina_object.externalIdentifier, version: updated_cocina_object.version) || VersionService.open?(druid: updated_cocina_object.externalIdentifier, version: updated_cocina_object.version))
+  raise 'Cannot version' if migrator.version? && !(VersionService.can_open?(druid: updated_cocina_object.externalIdentifier,
+                                                                            version: updated_cocina_object.version) || WorkflowStateService.active_version_wf?(druid: updated_cocina_object.externalIdentifier,
```
Comment (justinlittman) [inline bin/migrate-cocina:84]: Switched to WorkflowStateService since VersionService no longer exposes `open?`.

**Review by aaron-collier — APPROVED** [review]

**Pull request merged**